mm2src/coins/nft/nft_structs.rs

```diff
@@ -143,20 +184,17 @@ impl fmt::Display for Chain {
 impl FromStr for Chain {
     type Err = ParseChainTypeError;
 
-    #[inline]
+    /// Converts a string slice to a `Chain` enum.
+    /// This implementation is primarily used in the context of deserialization with Serde.
+    #[inline(always)]
     fn from_str(s: &str) -> Result<Chain, ParseChainTypeError> {
         match s {
-            "AVALANCHE" => Ok(Chain::Avalanche),
-            "avalanche" => Ok(Chain::Avalanche),
-            "BSC" => Ok(Chain::Bsc),
-            "bsc" => Ok(Chain::Bsc),
-            "ETH" => Ok(Chain::Eth),
-            "eth" => Ok(Chain::Eth),
-            "FANTOM" => Ok(Chain::Fantom),
-            "fantom" => Ok(Chain::Fantom),
-            "POLYGON" => Ok(Chain::Polygon),
-            "polygon" => Ok(Chain::Polygon),
-            _ => Err(ParseChainTypeError::UnsupportedChainType),
+            "AVALANCHE" | "avalanche" => Ok(Chain::Avalanche),
+            "BSC" | "bsc" => Ok(Chain::Bsc),
+            "ETH" | "eth" => Ok(Chain::Eth),
+            "FANTOM" | "fantom" => Ok(Chain::Fantom),
+            "POLYGON" | "polygon" => Ok(Chain::Polygon),
+            _ => Err(ParseChainTypeError::Chain),
         }
     }
 }
```

Comment:
`MmError<String>` is what we should avoid and provide the explicit rust error type (I mean use enum instead of returning strings).

But may be you are right, that I over complicated the enum type. Changed to:
```rust
/// Errors encountered when parsing a `Chain` from a string.
#[derive(Debug, Display)]
pub enum ParseChainTypeError {
    #[display(fmt = "The provided string does not correspond to any of the supported blockchain types.")]
    UnsupportedChainType,
}
```